ledger/participant-state-metrics/src/main/scala/com/daml/ledger/participant/state/metrics/MetricName.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.metrics
+
+import scala.language.implicitConversions
+
+final class MetricName(private val segments: Vector[String]) extends AnyVal {
+  def :+(segment: String): MetricName =
+    new MetricName(segments :+ segment)
+
+  override def toString: String =
+    segments.mkString(".")
+}
+
+object MetricName {
+
+  val DAML: MetricName = MetricName("daml")
```

Comment:
```suggestion
  val Daml: MetricName = MetricName("daml")
```

Reply:
Ugh, fine. 🙃

Reply:
Whoops, didn't realize that label was still on there. We can fix this in a subsequent PR.